File: bigdime-libraries/bigdime-hdfs-lib/src/main/java/io/bigdime/libs/hdfs/WebHdfsException.java
package io.bigdime.libs.hdfs;

public class WebHdfsException extends Exception {
	private static final long serialVersionUID = 1L;

	private int statusCode;

	public WebHdfsException(int _statusCode, String message) {
		super(message);
		this.statusCode = _statusCode;
	}

	public WebHdfsException(String message, Exception e) {
		super(message, e);
	}

	public int getStatusCode() {
		return statusCode;
	}
}

File: bigdime-libraries/src/main/scala/io/bigdime/libs/hdfs/WebHdfsException.scala
package io.bigdime.libs.hdfs

/**
  * Created by neejain on 2/1/17.
  */
case class WebHdfsException(message: String, cause: Throwable, statusCode: Int) extends Exception(message, cause) {
  def this(statusCode: Int, message: String) = {
    this(message, null, statusCode)
  }

  def this(message: String, cause: Throwable) = {
    this(message, cause, 0)
  }
}
